### DISCUSSION TITLE
Documentation + Container Host Support

### DISCUSSION THREAD

**Discussion by Erich-McMillan** [tianocore/containers#32] [General]
Hi all,

Awesome to see these containers available, I have been using the Ubuntu20 for development for a few days now have been pretty happy with it.

However, I had to make some modifications to the Ubuntu container to get it to build with a Windows host, specifically removing the .sh scripts. Seems this container is very oriented towards a specific workflow where a local Edk2 repo is shared from host to container via volume, on Windows my experience with volumes has been subpar due to performance. Usually, it is easier to just clone the repo in the container and remote in with VSCode->container. Maybe we can have a version of Ubuntu container that supports different development model?

Additionally, seems the Windows2022 container only works with Win11 host, I had to downgrade the servercore to 2019 to build it on my Win10 host. Maybe we can support VS2019 on Win10?

I'm happy to help improve documentation, but not 100% sure where to start, hoping this discussion can be the start of a plan to improve the documentation so new users can easily understand how to use these containers for development.

Thanks 😊

**Comment by cfernald** [comment]
Hey Erich,

First of all, thanks for the feedback!

Have you tried the Fedora image? This has been the one I have been primarily using and so making fixes to as I go. I'd be curious if this model works for you better so we can use that as a data point for other or future images. This is also the image we have already enabled in our CI and started vetting local use of in some of our repos. 

I have done a decent bit of experimenting on running in a Windows environment and the route I have actually used is cloning into WSL2 and running loading into a VS Code dev container from there. That avoids the terrible perf from Windows -> WSL (or WSL backed docker), but allows you to easily access the file system through the existing windows WSL shares. I have a todo item to document a good developer flow in this repo.

The Windows2022 is an issue I was not aware of. Would you mind opening an issue for this and assigning it to me?

**Reply by cfernald** [reply]
Created issue #34 for my above mentioned TODO

**Comment by Erich-McMillan** [comment]
Will try Fedora image in a bit here. I opened a new bug for the Win2022 issue: https://github.com/tianocore/containers/issues/35

**Comment by Erich-McMillan** [comment]
@cfernald  can build Fedora.
